.github/workflows/run-integration-tests.yml

```diff
@@ -43,11 +43,6 @@ jobs:
             agent: "MonologueAgent"
             embedding-model: "local"
             sandbox: "exec"
-          - name: CodeActAgent-py311-exec
-            python-version: "3.11"
-            agent: "CodeActAgent"
-            embedding-model: "none"
-            sandbox: "exec"
```

Comment:
Why are we removing this test case? I'd like to keep intercompatibility of agents and sandbox envs--is there a particular reason CodeAct can't work with exec?

Reply:
As discussed above: 

> CodeActAgent will fail with the exec integration task (well, and it is not designed to work with the exec box since it is not stateful). @rbren and I were thinking about deprecating ExecBox soon, so I removed it from the integration test for now.

CodeActAgent is not designed for execbox, even if we barely get it work there, it won't work well. e.g., CodeActAgent needs `source ~/.bashrc` to load plugins and tools (e.g., swe-agent primitives) to function properly. But **ExecBox is not stateful**, the commands/tools you registered in a previous execution is not available in the next `.execute` request, essentially make the whole thing not working.

I don't see a point of spending too much efforts in making CodeActAgent compatible with ExecBox, especially we are already considering deprecating it.

Reply:
I'll double-check to confirm that this PR will work well with DooD + SSHBox, and hopefully it will mitigate the issue of not being able to use ExecBox.

Reply:
You probably need to add a note to user doc that this agent doesn't work with execbox. Or even throw an exception.

Reply:
According to https://github.com/OpenDevin/OpenDevin/pull/1149 @foragerr, seems we cannot use sshbox for dogfood purpose?

Reply:
I'll do more extensive testing on DooD + SSHBox, and i think for dogfood in Github, we could also use `LocalBox` for this, which i do plan to test as well.

Reply:
That's a good point. LocalBox should be enough for dogfood CI to run.